src/create-client.ts

```diff
@@ -258,11 +261,31 @@ An authorization_code was supplied for a login which did not originate at the ap
   }
 
   async #refreshSession({ organizationId }: { organizationId?: string } = {}) {
+    if (this.#state.tag === "AUTHENTICATING") {
+      await this.#state.response;
+      return;
+    }
+
     const beginningState = this.#state;
-    this.#state = { tag: "AUTHENTICATING" };
 
+    this.#state = {
+      tag: "AUTHENTICATING",
+      response: this.#doRefresh({ organizationId, beginningState }),
+    };
+
+    await this.#state.response;
+    return;
+  }
+
+  async #doRefresh({
```

Comment:
There's starting to be some internal duplication between `#doRefresh` and `#handleCallback`, particularly around the setting of `this.#state` and `setSessionData`. No need to refactor yet, but I'd say something to keep an eye on.